sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -864,9 +849,15 @@ protected Symbol visitSubqueryExpression(SubqueryExpression node, ExpressionAnal
              *
              * However, we support a single column RowType through the SingleColumnTableType.
              */
-            Field field = fields.get(0);
-            SingleColumnTableType singleColumnTableType = new SingleColumnTableType(field.valueType());
-            return new SelectSymbol(relation, singleColumnTableType);
+            DataType innerType = fields.get(0).valueType();
+            SingleColumnTableType dataType = new SingleColumnTableType(innerType);
+            final SelectSymbol.ResultType resultType;
+            if (context.isArrayExpression(node)) {
```

Comment:
What do you mean by "this wouldn't be safe"?

In an earlier version, I set a flag on the `SubqueryExpression` but that would be a parser-level change which is kind of out of place here. We could also generate an intermediate expression which would be similar to a flag.